pkg/core/blockchain.go

```diff
@@ -601,9 +603,12 @@ func (bc *Blockchain) storeBlock(block *block.Block) error {
 			}
 		}
 	}
+
 	bc.lock.Lock()
 	defer bc.lock.Unlock()
 
+	bc.lastBatch = cache.dao.store.GetBatch()
```

Comment:
NACK. It has to be less invasive, 99,99% of the time we don't care and wasting CPU and memory at this spot is not what we want.